OpenRA.Mods.Common/Scripting/Properties/PlayerProperties.cs

```diff
@@ -32,5 +33,24 @@ public Actor[] GetGroundAttackers()
 				.Where(a => a.Owner == Player && !a.IsDead && a.IsInWorld && a.HasTrait<Mobile>())
 				.ToArray();
 		}
+
+		[Desc("Returns an array of actors of the specified type in the specified maximum amount.")]
+		public Actor[] GetActorsByType(LuaTable types)
+		{
+			var result = new List<Actor>();
+
+			foreach (var type in types)
+			{
+				ActorInfo ai;
+				if (!Context.World.Map.Rules.Actors.TryGetValue(type.Key.ToString(), out ai))
+					throw new LuaException("Unknown actor type '{0}'".F(type.Key.ToString()));
+
+				result.AddRange(Player.World.ActorMap.ActorsInWorld()
+					.Where(actor => actor.Owner == Player && !actor.IsDead && actor.IsInWorld && actor.Info.Name == ai.Name)
+					.Take(int.Parse(type.Value.ToString())));
```

Comment:
This is a bit weird. It says type, but also means count. Also why are you making the decision about the size of the returning array here? Just spit out the everything and have the Lua script decide on how many to use from the array. The syntax from:

```lua
AutoCreateTeams = 
{
	{ { ['e1'] = 1,  ['e3'] = 3 }, AtkRoute2 },
	{ { ['e1'] = 3,  ['e3'] = 1 }, AtkRoute2 },
	{ { ['e3'] = 4 }             , AtkRoute1 },
	{ { ['e1'] = 4 }             , AtkRoute1 },
	{ { ['bggy'] = 1 }           , AtkRoute1 },
	{ { ['bggy'] = 1 }           , AtkRoute2 },
	{ { ['ltnk'] = 1 }           , AtkRoute1 },
	{ { ['ltnk'] = 1 }           , AtkRoute2 }
}
```

with the hard-coded result size values is not very obvious and seems to overcomplicate things.